### PR TITLE
Fix 500 Error on Problems PUT

### LINKED ISSUE #22: 500 Error on PUT Problems
Issuing a PUT request on `/problems/:problemid` results in a 500 error - upon further investigation, I've determined this is because the controller reassigns the value of the `problemId` from the MongoID which is expected on the request to the computed Platform-Problem hash.

### PR DESCRIPTION
**Pull request by idodin**
Resolves #22 

Fixes error where problem hash masked the Mongo ID expected in the
request parameter, resulting in a 500 error on the PUT
/problems/:problemId endpoint.

Renames retrieved problemId from header to be more explicit about
variable usage.

### CODE REVIEW

**Comment by idodin** [comment]
> instead of changing the name of the param to `mongoId` everywhere, why not change the name of the variable in the update function to something more fitting like `platformProblemIdHash`?

That's fair, I agree that's more descriptive

File: src/controllers/problem.ts [hunk 1]
```diff
@@ -90,7 +90,7 @@ const problemController = {
             res.status(statusCodes.MISSING_PARAMS).json(errors.formatWith(errorMessage).array()[0]);
         } else {
             try {
-                const { problemId } = req.params;
+                const problemId: string  = req.params.problemId;
```
Comment (MohamedBeydoun) [inline src/controllers/problem.ts:93]: we should keep the deconstructed format ( const { problemId } = req.params;) to stay consistent with the rest of the code across the microservices

File: src/controllers/problem.ts [hunk 2]
```diff
@@ -133,7 +132,7 @@ const problemController = {
             res.status(statusCodes.MISSING_PARAMS).json(errors.formatWith(errorMessage).array()[0]);
         } else {
             try {
-                const { problemId } = req.params;
+                const problemId = req.params.problemId;
```
Comment (MohamedBeydoun) [inline src/controllers/problem.ts:135]: same as above

File: src/controllers/problem.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+import {Request, Response} from "express";
+import {problemDBInteractions} from "../database/interactions/problem";
+import {IProblemModel, Problem} from "../database/models/problem";
+import {IProblem} from "../interfaces/IProblem";
+import {validationResult} from "express-validator/check";
+import {errorMessage} from "../config/errorFormatter";
+import {statusCodes} from "../config/statusCodes";
+import {calculateProblemHash} from "../util/hash";
+import {problemSetDBInteractions} from "../database/interactions/problemSet";
+import {IProblemSetModel} from "../database/models/problemSet";
```
Comment (ValerianClerc) [inline src/controllers/problem.ts:10]: Can you fix the linting that your code editor did here?

Side-note: we should set up eslint or prettier at some point on the backend repos lol

**Pull request merged**